src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -2,18 +2,197 @@
 
 public class MyHashMap<K, V> implements MyMap<K, V> {
 
+    private static final int DEFAULT_CAPACITY = 16;
+    private static final float DEFAULT_LOAD_FACTOR = 0.75f;
+    private Node<K, V>[] table;
+    private int size;
+
+    @SuppressWarnings("unchecked")
+    public MyHashMap() {
+        this.table = (Node<K,V>[])new Node[DEFAULT_CAPACITY];
+    }
+
     @Override
     public void put(K key, V value) {
-
+        resizeIfNeeded();
+        int index = calculateIndex(key);
+        if (table[index] == null) {
+            Node<K, V> newNode = new Node<>(key, value);
+            table[index] = newNode;
+        } else {
+            Node<K, V> current = table[index];
+            while (current.next != null) {
+                if (isEqual(key, current.key)) {
+                    current.value = value;
+                    return;
+                }
+                current = current.next;
+            }
+            if (isEqual(key, current.key)) {
+                current.value = value;
+                return;
+            }
+            current.next = new Node<>(key, value);
+        }
+        size++;
     }
 
     @Override
     public V getValue(K key) {
+        for (Node<K, V> node : table) {
+            if (node != null) {
+                if (node.next == null) {
+                    if (isEqual(key, node.key)) {
+                        return node.value;
+                    }
+                } else {
+                    Node<K, V> current = node;
+                    do {
+                        if (isEqual(key, current.key)) {
+                            return current.value;
+                        }
+                        current = current.next;
+                    } while (current != null);
+                }
+            }
+        }
```

Comment:
Instead of it, you can get first node in bucket by index(you have method calculate index) and then find Node you need